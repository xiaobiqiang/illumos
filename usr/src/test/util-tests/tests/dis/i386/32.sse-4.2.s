/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2016 Joyent, Inc.
 */

/*
 * Test SSE 4.2 related instructions
 */

.text
.align 16
.globl libdis_test
.type libdis_test, @function
libdis_test:
	crc32b		%al, %ebx
	crc32b		(%eax), %ebx
	crc32w		%ax, %ebx
	crc32w		(%eax), %ebx
	crc32l		%eax, %ebx
	crc32l		(%eax), %ebx
	pcmpestri	$0x23, %xmm0, %xmm1
	pcmpestri	$0x23, (%eax), %xmm1
	pcmpestrm	$0x23, %xmm0, %xmm1
	pcmpestrm	$0x23, (%eax), %xmm1
	pcmpgtq		%xmm0, %xmm1
	pcmpgtq		(%eax), %xmm1
	pcmpistri	$0x23, %xmm0, %xmm1
	pcmpistri	$0x23, (%eax), %xmm1
	pcmpistrm	$0x23, %xmm0, %xmm1
	pcmpistrm	$0x23, (%eax), %xmm1
.size libdis_test, [.-libdis_test]
