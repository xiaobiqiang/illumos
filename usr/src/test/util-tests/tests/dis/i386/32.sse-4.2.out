    libdis_test:      f2 0f 38 f0 d8     crc32b %al,%ebx
    libdis_test+0x5:  f2 0f 38 f0 18     crc32b (%eax),%ebx
    libdis_test+0xa:  66 f2 0f 38 f1 d8  crc32w %ax,%ebx
    libdis_test+0x10: 66 f2 0f 38 f1 18  crc32w (%eax),%ebx
    libdis_test+0x16: f2 0f 38 f1 d8     crc32l %eax,%ebx
    libdis_test+0x1b: f2 0f 38 f1 18     crc32l (%eax),%ebx
    libdis_test+0x20: 66 0f 3a 61 c8 23  pcmpestri $0x23,%xmm0,%xmm1
    libdis_test+0x26: 66 0f 3a 61 08 23  pcmpestri $0x23,(%eax),%xmm1
    libdis_test+0x2c: 66 0f 3a 60 c8 23  pcmpestrm $0x23,%xmm0,%xmm1
    libdis_test+0x32: 66 0f 3a 60 08 23  pcmpestrm $0x23,(%eax),%xmm1
    libdis_test+0x38: 66 0f 38 37 c8     pcmpgtq %xmm0,%xmm1
    libdis_test+0x3d: 66 0f 38 37 08     pcmpgtq (%eax),%xmm1
    libdis_test+0x42: 66 0f 3a 63 c8 23  pcmpistri $0x23,%xmm0,%xmm1
    libdis_test+0x48: 66 0f 3a 63 08 23  pcmpistri $0x23,(%eax),%xmm1
    libdis_test+0x4e: 66 0f 3a 62 c8 23  pcmpistrm $0x23,%xmm0,%xmm1
    libdis_test+0x54: 66 0f 3a 62 08 23  pcmpistrm $0x23,(%eax),%xmm1
